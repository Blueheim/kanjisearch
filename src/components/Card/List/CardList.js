import React from 'react';
import Card from '../Card';

const CardList = ({ kanjis }) => {
  return (
    <div>
      {kanjis.map((kanji, i) => {
        return (
          <Card
            key={i}
            id={kanjis[i].id}
            meaning={kanjis[i].meaning.english}
            character={kanjis[i].character}
            kunyomi={kanjis[i].kunyomi}
            onyomi={kanjis[i].onyomi}
          />
        );
      })}
    </div>
  );
};

export default CardList;
