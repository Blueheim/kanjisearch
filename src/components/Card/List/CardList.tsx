import React from 'react';
import Card from '../Card';
import { IKanji } from '../../../types/kanji';

import classes from './CardList.module.scss';

const CardList = ({ kanjis }: { kanjis: IKanji[] }) => {
  return (
    <section className="l-section l-section--simple">
      <div className={[classes.Cardlist__content, 'm-fx-c-sh', 'm-pd-xt'].join(' ')}>
        {kanjis.map((kanji, i) => {
          return (
            <Card
              key={i}
              id={kanji.id}
              meaning={kanji.meaning.english}
              character={kanji.character}
              kunyomi={kanji.kunyomi}
              onyomi={kanji.onyomi}
            />
          );
        })}
      </div>
    </section>
  );
};

export default CardList;
