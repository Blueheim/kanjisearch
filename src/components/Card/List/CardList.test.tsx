import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

describe('<CardList />', () => {
  const kanjis = [
    {
      id: 1,
      character: 'X',
      meaning: ['blabla'],
      kunyomi: ['ytyt', 'fgfdgd'],
      onyomi: ['ffdf', 'fsdf'],
    },
  ];

  const wrapper = shallow(<CardList kanjis={kanjis} />);

  it('expect to render CardList component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
