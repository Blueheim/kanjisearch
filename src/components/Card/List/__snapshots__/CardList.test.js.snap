// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> expect to render CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    kanjis={
      Array [
        Object {
          "character": "X",
          "id": 1,
          "kunyomi": Array [
            "ytyt",
            "fgfdgd",
          ],
          "meaning": Array [
            "blabla",
          ],
          "onyomi": Array [
            "ffdf",
            "fsdf",
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Cardlist__content m-fx-c-sh m-pd-xt"
      >
        <Card
          character="X"
          id={1}
          kunyomi={
            Array [
              "ytyt",
              "fgfdgd",
            ]
          }
          onyomi={
            Array [
              "ffdf",
              "fsdf",
            ]
          }
        />
      </div>,
      "className": "l-section l-section--simple",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            character="X"
            id={1}
            kunyomi={
              Array [
                "ytyt",
                "fgfdgd",
              ]
            }
            onyomi={
              Array [
                "ffdf",
                "fsdf",
              ]
            }
          />,
        ],
        "className": "Cardlist__content m-fx-c-sh m-pd-xt",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "character": "X",
            "id": 1,
            "kunyomi": Array [
              "ytyt",
              "fgfdgd",
            ],
            "meaning": undefined,
            "onyomi": Array [
              "ffdf",
              "fsdf",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Cardlist__content m-fx-c-sh m-pd-xt"
        >
          <Card
            character="X"
            id={1}
            kunyomi={
              Array [
                "ytyt",
                "fgfdgd",
              ]
            }
            onyomi={
              Array [
                "ffdf",
                "fsdf",
              ]
            }
          />
        </div>,
        "className": "l-section l-section--simple",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              character="X"
              id={1}
              kunyomi={
                Array [
                  "ytyt",
                  "fgfdgd",
                ]
              }
              onyomi={
                Array [
                  "ffdf",
                  "fsdf",
                ]
              }
            />,
          ],
          "className": "Cardlist__content m-fx-c-sh m-pd-xt",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "character": "X",
              "id": 1,
              "kunyomi": Array [
                "ytyt",
                "fgfdgd",
              ],
              "meaning": undefined,
              "onyomi": Array [
                "ffdf",
                "fsdf",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
